include/libsurvive/survive_api.h

```diff
@@ -0,0 +1,63 @@
+#include "survive_types.h"
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+	struct SurviveAsyncContext;
+	typedef struct SurviveAsyncContext SurviveAsyncContext;
+
+	/***
+	 * Initialize a new instance of an async context -- mirrors survive_init
+	 * @return Pointer to the async context
+	 */
+	SURVIVE_EXPORT SurviveAsyncContext *survive_async_init(int argc, char *const *argv);
```

Comment:
Possibly rename this to 'optionsc' / optionsv'.  It doesn't have to be argc / argv and seems like a decent way to pass in config data. I do like, from a practical standpoint, that if you hook it right into the program arguments, you get consistent behavior across all the tools.

But I also think we need to do the right thing if the user passes in 0, 0 here; and expose the config functions. Possibly have an '..._init_with_options' or something that takes in the string array?

Reply:
I think just accepting 0,0 should be sufficient.  Some applications let you pass additional parameters to subsystems that have command-line parameters themselves.  Additionally, because the way we parse parameters isn't too squirrely, it is easy for applications to pass in their own options.  But, I could see the value in bringing back the two-step initialization and allowing parameters to be set between the first and second stages.